manifests/configure.pp

```diff
@@ -4,193 +4,133 @@
-  validate_string($apache_pass)
-  validate_string($haproxy_stats_url)
-  validate_string($java_app_type)
-  validate_string($java_app_subtype)
-  validate_string($java_jvm_name)
-  validate_integer($memcached_port)
-  validate_string($mysql_user)
-  validate_string($mysql_pass)
-  validate_string($mysql_host)
-  validate_integer($mysql_port)
-  validate_string($nginx_stats_url)
-  validate_string($nginx_plus_stats_url)
+  if $args == undef {
+    $command = "/opt/spm/bin/setup-sematext \\
+      --monitoring-token ${ monitoring_token } \\
+      --infra-token ${ infra_token } \\
+      --agent-type ${ agent_type } \\
+      --app-type ${ app_type }"
 
-  if $spm_token == undef {
-    fail("spm_token is mandatory")
-  }
-  if $spm_type == undef {
-    fail("spm_token is mandatory")
-  }
+    $config = "/opt/spm/spm-monitor/conf/spm-monitor-config-${ monitoring_token }-${ app_name }.properties"
 
-  case $spm_type {
-    'apache': {
-      spm_monitor::configure::apache { "${spm_token}":
-        spm_token => $spm_token,
-        stats_url => $apache_stats_url,
-        user => $apache_user,
-        pass => $apache_pass
-      }
-    }
-    'haproxy': {
-      spm_monitor::configure::haproxy { "${spm_token}":
-        spm_token => $spm_token,
-        stats_url => $haproxy_stats_url
-      }
-    }
-    'memcached': {
-      spm_monitor::configure::memcached { "${spm_token}":
-        spm_token => $spm_token,
-        port => $memcached_port
-      }
-    }
-    'mysql': {
-      spm_monitor::configure::mysql { "${spm_token}":
-        spm_token => $spm_token,
-        user => $mysql_user,
-        pass => $mysql_pass,
-        host => $mysql_host,
-        port => $mysql_port
-      }
-    }
-    'nginx': {
-      spm_monitor::configure::nginx { "${spm_token}":
-        spm_token => $spm_token,
-        stats_url => $nginx_stats_url,
-        user => $nginx_user,
-        pass => $nginx_pass
-      }
+    exec { "${command}":
+      creates => $config,
     }
-    'nginx-plus': {
-      spm_monitor::configure::nginxplus { "${spm_token}":
-        spm_token => $spm_token,
-        stats_url => $nginx_plus_stats_url
-      }
-    }
-    'javaagent': {
-      spm_monitor::configure::javaagent { "${spm_token}-${java_app_type}-${java_jvm_name}":
-        spm_token => $spm_token,
-        app_type => $java_app_type,
-        app_subtype => $java_app_subtype,
-        jvm_name => $java_jvm_name,
-        tracing => $java_tracing
+  }
+  else {
+    $extra_args = $args
+      .map |$key, $value| { "--${key} ${value}" }
+      .join(' ')
+
+    if !has_key($args, 'app_subtype') {
+      $command = "/opt/spm/bin/setup-sematext \\
+        --monitoring-token ${ monitoring_token } \\
+        --infra-token ${ infra_token } \\
+        --agent-type ${ agent_type } \\
+        --app-type ${ app_type } \\
+        ${extra_args}"
+
+      $config = "/opt/spm/spm-monitor/conf/spm-monitor--config-${ monitoring_token }-${ app_name }.properties"
```

Comment:
we check for `app_subtype` in the condition above and use `app_name` in config.

Reply:
we check for **not** having 'app_subtype' above.